fds.go

```diff
@@ -278,7 +279,7 @@ func (f *Fds) closeInherited() {
 }
 
 func unlinkUnixSocket(path string) error {
-	if strings.HasPrefix(path, "@") {
+	if runtime.GOOS != "darwin" && strings.HasPrefix(path, "@") {
```

Comment:
Could also use build flags for this, but it doesn't seem worthwhile to me for this

Reply:
GOOS seems fine, maybe even change it to `runtime.GOOS == "linux"`?